### ISSUE TITLE
Arguments to ExposedThing.register() 

### ISSUE THREAD

**Issue by zolkis** [w3c/wot-scripting-api#87]
@mmccool suggested the `register()` method should accept a `ttl` (time to live) argument, and perhaps `autoupdate` boolean flag.

This would mean to change the API from 
`Promise<void> register(optional USVString directory);`  
to 
```javascript
Promise<void> register(optional RegisterOptions options);

dictionary RegisterOptions {
    USVString directory;
    double ttl;
    boolean autoupdate;
}
```

**Comment by zolkis** [comment]
If TTL is needed, shouldn't it be a property in the TD, rather than a deployment parameter that is not reflected in the TD?

**Comment by mkovatsc** [comment]
"TTL" is cache-control metadata for a TD document (cf. `Cache-Contol` header field or CoAP's `Max-Age` option). Its value depends on the code that created and potentially modifies the ExposedThing. Thus, it makes sense to pass it along with `register`. Directories also need to update this value just like a cache, so they would also store it as metadata alongside the TD.

Also see lifetime of the [CoRE Resource Directory](https://tools.ietf.org/html/draft-ietf-core-resource-directory-12).

**Comment by mkovatsc** [comment]
We need a flexible object that we can pass to `register()` -- or rather `expose()`, see #75 -- so that we have extension points. It would somewhat be a counter part to the filter object for `discover()`.

One requirement is, that the object must support to take the fields expected by the CoRE RD (directly or in an abstracted form), as we also use them in the Thing Directory.

**Comment by zolkis** [comment]
This has been sufficiently addressed in later PRs (such as #128 and later).
